File: hw/calle/u2/hw19Inheritance/FarmsystemPython/farmsystem_model/cage.py
from .location import Location
from typing import Self

class Cage:
    def __init__(self, id: int, description: str, type: int, location: Location):
        self._id = id
        self._description = description
        self._type = type
        self._location = location

    def __str__(self) -> str:
        return f"Cage{{id={self.id}, description={self.description}, type={self.type}, location={self.location}}}"

    @property
    def id(self) -> int:
        return self._id

    @id.setter
    def id(self, value: int):
        self._id = value

    @property
    def description(self) -> str:
        return self._description

    @description.setter
    def description(self, value: str):
        self._description = value

    @property
    def type(self) -> int:
        return self._type

    @type.setter
    def type(self, value: int):
        self._type = value

    @property
    def location(self) -> Location:
        return self._location

    @location.setter
    def location(self, value: Location):
        self._location = value

File: hw/toapanta/u2/hw19inherence/javaScriptFarmSystem/model/Cage.js
<<<<<<< HEAD
class Cage {
    constructor(id, description, type, location) {
        this.id = id;
        this.description = description;
        this.type = type;
        this.location = location;
    }

    toString() {
        return `Cage\nid.: ${this.id}\ndescription: ${this.description}\ntype: ${this.type}`;
    }

    getId() {
        return this.id;
    }

    setId(id) {
        this.id = id;
    }

    getDescription() {
        return this.description;
    }

    setDescription(description) {
        this.description = description;
    }

    getType() {
        return this.type;
    }

    setType(type) {
        this.type = type;
    }
}

=======
class Cage {
    constructor(id, description, type, location) {
        this.id = id;
        this.description = description;
        this.type = type;
        this.location = location;
    }

    toString() {
        return `Cage\nid.: ${this.id}\ndescription: ${this.description}\ntype: ${this.type}`;
    }

    getId() {
        return this.id;
    }

    setId(id) {
        this.id = id;
    }

    getDescription() {
        return this.description;
    }

    setDescription(description) {
        this.description = description;
    }

    getType() {
        return this.type;
    }

    setType(type) {
        this.type = type;
    }
}

>>>>>>> 203b676d8f105a34d549ec251a59c11aa2c57532
module.exports = Cage;